terraform_files/equinix-ci-machine/variables_equinix.tf

```diff
@@ -0,0 +1,45 @@
+variable "global_project_id" {
+  type        = string
+  description = "Project in which the devices will be created"
+}
+
+variable "global_facilities" {
+  type        = list(string)
+  default     = ["any"]
+  description = "Metro in which the devices will be created"
+}
+
+variable "global_operating_system" {
+  type        = string
+  default     = "rocky_8"
+  description = "Operating system that will be installed in the created devices"
+}
+
+variable "global_tags" {
+  type        = list(string)
+  default     = []
+  description = "Global tags associated to the created devices"
+}
+
+variable "ssh_private_key_path" {
+  type        = string
+  description = "Path to SSH private key authorized to connect on the created devices"
+}
+
+variable "hostname_prefix" {
+  type        = string
+  description = "Prefix as used as hostname for the created devices"
+}
+
+variable "devices" {
+  description = "The total configuration, List of Objects/Dictionary"
+  type = list(
+    object({
+      plan             = string
+      facilities       = optional(list(string), null)
+      operating_system = optional(string, null)
+      project_id       = optional(string, null)
```

Comment:
on what cases we'll want those to be optional/null?

Reply:
yeah, you're right, we can offload most of the templating to jinja/ansible, it would be more maintainable than mixing both.

Reply:
ok I moved the heavy lifting in ansible, it's cleaner I think now, thanks!